middleware/assert_test.go

```diff
@@ -124,7 +124,54 @@ func TestAssertFilterType(t *testing.T){
 			})
 		})
 
-		Convey("When a filter with given ID in the datastore is found, but type is not 'flexible'", func(){
+		Convey("When a filter output with given ID in the datastore is found, but type is not 'flexible'", func() {
+			datastoreMock := mock.NewDataStore().Mock
+			datastoreMock.GetFilterOutputFunc = func(ctx context.Context, filterID string) (*models.Filter, error) {
+				return &models.Filter{
+					Type: "foobar",
+					Dataset: &models.Dataset{
+						Version: 2,
+					},
+					ID: filterID,
+				}, nil
+			}
+
+			Convey("When an incoming request pases through the asert.FilterOutputType middleware", func() {
```

Comment:
Would it be possible to add a test for when the filter output is not flexible but is still checked by `assert.FilterOutputFilterType` and therefore not forwarded to the filter-flex-api?

Reply:
@jackrehaag line 1231 of `api/filter_test.go` should do what you seek.

Reply:
I think it would be best to have both cases covered in `assert_test.go` that way we're covered if the filter in question is modified or deleted.

Reply:
Np. Added tests that cover both cases, with the appropriate conditions in `assert_test.go`. I will keep this conversation resolved for that reason